### PR TITLE
feat(richtext-lexical)!: Lexical i18n

### PR DESCRIPTION
**Pull request by AlessioGr**
## Description

**BREAKING:** All slash menu `displayName` properties have been renamed to `label`.

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [ ] Chore (non-breaking change which does not add functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Change to the [templates](https://github.com/payloadcms/payload/tree/main/templates) directory (does not affect core functionality)
- [ ] Change to the [examples](https://github.com/payloadcms/payload/tree/main/examples) directory (does not affect core functionality)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/payload/src/fields/config/sanitize.ts [hunk 1]
```diff
@@ -44,8 +53,41 @@ export const sanitizeFields = ({
     }
 
     // Make sure that the richText field has an editor
-    if (field.type === 'richText' && !field.editor && config.editor) {
-      field.editor = config.editor
+    if (field.type === 'richText') {
```
Comment (DanRibbens) [inline packages/payload/src/fields/config/sanitize.ts:56]: All of the editor configuration logic should be done in a new file. These changes from L56 - L90 are hard to read.

File: packages/payload/src/fields/config/sanitize.ts [hunk 2]
```diff
@@ -44,8 +53,41 @@ export const sanitizeFields = ({
...
+      if (!field.editor && config.editor) {
+        field.editor = config.editor
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { i18nServerToUse, serverEditor } = field?.editor?.i18nServer
+        ? { i18nServerToUse: field?.editor?.i18nServer, serverEditor: field?.editor }
+        : { i18nServerToUse: config?.editor?.i18nServer, serverEditor: config?.editor }
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { clientEditor, i18nClientToUse } = field?.editor?.i18nClient
```
Comment (DanRibbens) [inline packages/payload/src/fields/config/sanitize.ts:66]: Should this be skipped if initializing Payload with `localOnly`?

I'm unclear on the move to i18nServer vs i18nClient.

**Review by DanRibbens — COMMENTED** [review]
Just some clarifying questions, looks good overall.

**Comment by AlessioGr** [comment]
Closing in favor of #6542. i18n won't make it to 2.0

**Pull request closed without merging**